### PR TITLE
Better error message for test that should panic but passes

### LINKED ISSUE #1463: Better error message for test that should panic but passes
https://t.me/starknet_foundry/699
The place to be changed: https://github.com/foundry-rs/starknet-foundry/blob/2d6c59c96f11968933a90d3c5034d6568298afe7/crates/forge-runner/src/test_case_summary.rs#L206

### PR DESCRIPTION
**Pull request by Draggu**
Closes #1463

## Introduced changes

Added better error message for test that should panic but passes.

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 #### Added
 
+- Error message for test that should panic but passes
```
Comment (drknzz) [inline CHANGELOG.md:14]:
```suggestion
- Error message for tests that should panic but pass
```

File: crates/forge-runner/src/test_case_summary.rs [hunk 2]
```diff
@@ -257,13 +257,32 @@ fn build_readable_text(data: &Vec<Felt252>) -> Option<String> {
...
+        RunResultValue::Success(_) => match expectation {
+            ExpectedTestResult::Panics(panic_expectation) => match panic_expectation {
+                ExpectedPanicValue::Exact(panic_data) => {
+                    let panic_string = join_short_strings(panic_data);
+
+                    Some(format!(
+                        "\n    Expected to panic with data\n    Expected:  {panic_data:?} ({panic_string})\n"
+                    ))
+                }
+                ExpectedPanicValue::Any => Some("\n    Expected to panic".into()),
```
Comment (drknzz) [inline crates/forge-runner/src/test_case_summary.rs:282]: This isn't tested i believe

**Review by drknzz — CHANGES_REQUESTED** [review]

File: crates/forge-runner/src/test_case_summary.rs [hunk 3]
```diff
@@ -257,13 +257,32 @@ fn build_readable_text(data: &Vec<Felt252>) -> Option<String> {
...
 fn extract_result_data(run_result: &RunResult, expectation: &ExpectedTestResult) -> Option<String> {
     match &run_result.value {
-        RunResultValue::Success(data) => build_readable_text(data),
+        RunResultValue::Success(_) => match expectation {
+            ExpectedTestResult::Panics(panic_expectation) => match panic_expectation {
+                ExpectedPanicValue::Exact(panic_data) => {
+                    let panic_string = join_short_strings(panic_data);
+
+                    Some(format!(
+                        "\n    Expected to panic with data\n    Expected:  {panic_data:?} ({panic_string})\n"
```
Comment (piotmag769) [inline crates/forge-runner/src/test_case_summary.rs:279]:
```suggestion
                        "\n    Expected to panic but didn't\n    Expected panic data:  {panic_data:?} ({panic_string})\n"
```
[nit]

File: crates/forge-runner/src/test_case_summary.rs (hunk 2 above)
Comment (piotmag769) [inline crates/forge-runner/src/test_case_summary.rs:282]:
```suggestion
                ExpectedPanicValue::Any => Some("\n    Expected to panic but didn't".into()),
```
[nit]

**Review by piotmag769 — COMMENTED** [review]
Good job and congrats on your first PR! 🎉 
Add the requested test and it's good to go imo

File: crates/forge-runner/src/test_case_summary.rs [hunk 4]
```diff
@@ -257,13 +257,32 @@ fn build_readable_text(data: &Vec<Felt252>) -> Option<String> {
...
+        RunResultValue::Success(data) => match expectation {
+            ExpectedTestResult::Panics(panic_expectation) => match panic_expectation {
+                ExpectedPanicValue::Exact(panic_data) => {
+                    let panic_string = join_short_strings(panic_data);
+
+                    Some(format!(
+                        "\n    Expected to panic but didn't\n    Expected panic data:  {panic_data:?} ({panic_string})\n"
+                    ))
+                }
+                ExpectedPanicValue::Any => Some("\n    Expected to panic but didn't".into()),
```
Comment (drknzz) [inline crates/forge-runner/src/test_case_summary.rs:282]: Probably should end with newline for consistency

File: crates/forge/tests/e2e/running.rs [hunk 5]
```diff
@@ -843,42 +843,60 @@ fn should_panic() {
...
         Running 0 test(s) from src/
-        Running 6 test(s) from tests/
-        [PASS] tests::should_panic_test::should_panic_no_data [..]
-
+        Running 8 test(s) from tests/
+        [FAIL] tests::should_panic_test::expected_panic_but_didnt
+        
+        Failure data:
+            Expected to panic but didn't
+        [PASS] tests::should_panic_test::should_panic_check_data (gas: ~1)
```
Comment (drknzz) [inline crates/forge/tests/e2e/running.rs:861]:
```suggestion
        [PASS] tests::should_panic_test::should_panic_check_data [..]
```
Let's not assert gas in here

Comment (drknzz) [reply crates/forge/tests/e2e/running.rs]: Let's do it in the whole test*, sorry for not clarifying!

File: crates/forge/tests/e2e/running.rs [hunk 6]
```diff
@@ -843,42 +843,61 @@ fn should_panic() {
     let snapbox = test_runner();
 
     let output = snapbox.current_dir(&temp).assert().code(1);
+
     assert_stdout_contains!(
```
Comment (piotmag769) [inline crates/forge/tests/e2e/running.rs:847]: I know you prob copied the output you got (and it's completely fine), just FYI: this macro doesn't check the order in which lines are printed

**Review by piotmag769 — APPROVED** [review]

**Review by drknzz — APPROVED** [review]
Nicely done, good job 🔥 👍

**Pull request merged**